prov/sockets/src/sock_cntr.c

```diff
@@ -210,8 +210,9 @@ void sock_cntr_inc(struct sock_cntr *cntr)
 {
 	pthread_mutex_lock(&cntr->mut);
 	atomic_inc(&cntr->value);
-	if (atomic_get(&cntr->value) >= atomic_get(&cntr->threshold))
-		pthread_cond_signal(&cntr->cond);
+	/* if(atomic_get(&cntr->waiting)) { */
```

Comment:
Is there a reason for commenting this condition? Since this is in the critical path, it makes sense to call the signal() only if somebody is waiting on the counter. Also, I had noticed some performance penalties if we call signal() always.